### PR TITLE
Remove internal rmw_connext_cpp headers from public API.

### PR DESCRIPTION
**Pull request by hidmic**
Precisely what the title says. Spin off from #446.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Please use separate commits to move and modify files (and then merge without squashing afterwards) to keep the history accessible.

**Comment by hidmic** [comment]
CI up to `rmw_connext_cpp` only:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11732/)
* Linux-aarch64 (no RTI Connext, excluded)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9570/)
* Windows [Build Status](https://ci.ros2.org/job/ci_windows/11670/)

**Comment by hidmic** [comment]
@dirk-thomas PTAL

File: rmw_connext_cpp/src/rmw_publisher.cpp [hunk 1]
```diff
@@ -28,9 +28,10 @@
 
 #include "rmw_connext_cpp/identifier.hpp"
 
+#include "connext_static_publisher_info.hpp"
 #include "process_topic_and_service_names.hpp"
 #include "type_support_common.hpp"
-#include "rmw_connext_cpp/connext_static_publisher_info.hpp"
+
```
Comment (dirk-thomas) [inline rmw_connext_cpp/src/rmw_publisher.cpp:34]: Nitpick: delete this empty line?

Comment (hidmic) [reply rmw_connext_cpp/src/rmw_publisher.cpp:34]: Yes! An oversight. See https://github.com/ros2/rmw_connext/pull/447/commits/ed082c3c8b4c867d8845e82be184dfbdef6f54dd

**Comment by hidmic** [comment]
I'll rebase to make sure only two commits go into `master`.

**Comment by hidmic** [comment]
Alright, going in.

**Pull request merged**